### ISSUE TITLE
PostgreSQL 9.6 Install Failure

### ISSUE THREAD

**Issue by jliles** [system76/pop-distro#43]
**Reporter info**

```
 * Distribution - Pop!_OS 
No LSB modules are available.
Distributor ID:	Ubuntu
Description:	Ubuntu 17.04
Release:	17.04
Codename:	zesty

 * Related Application and/or Package Version - postgresql
```

**Actual issue**
```sudo apt install postgresql
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following additional packages will be installed:
  libpq5 postgresql-9.6 postgresql-client-9.6 postgresql-client-common
  postgresql-common postgresql-contrib-9.6 sysstat
Suggested packages:
  postgresql-doc locales-all postgresql-doc-9.6 libdbd-pg-perl isag
The following NEW packages will be installed:
  libpq5 postgresql postgresql-9.6 postgresql-client-9.6
  postgresql-client-common postgresql-common postgresql-contrib-9.6 sysstat
0 upgraded, 8 newly installed, 0 to remove and 0 not upgraded.
Need to get 5,035 kB of archives.
After this operation, 20.3 MB of additional disk space will be used.
Do you want to continue? [Y/n] 
Get:1 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 libpq5 amd64 9.6.3-0ubuntu0.17.04 [79.4 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 postgresql-client-common all 179 [29.0 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 postgresql-client-9.6 amd64 9.6.3-0ubuntu0.17.04 [887 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 postgresql-common all 179 [157 kB]
Get:5 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 postgresql-9.6 amd64 9.6.3-0ubuntu0.17.04 [3,114 kB]
Get:6 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 postgresql all 9.6+179 [5,684 B]
Get:7 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 postgresql-contrib-9.6 amd64 9.6.3-0ubuntu0.17.04 [491 kB]
Get:8 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 sysstat amd64 11.4.3-1 [272 kB]
Fetched 5,035 kB in 1s (4,333 kB/s)
Preconfiguring packages ...
Selecting previously unselected package libpq5:amd64.
(Reading database ... 214441 files and directories currently installed.)
Preparing to unpack .../0-libpq5_9.6.3-0ubuntu0.17.04_amd64.deb ...
Unpacking libpq5:amd64 (9.6.3-0ubuntu0.17.04) ...
Selecting previously unselected package postgresql-client-common.
Preparing to unpack .../1-postgresql-client-common_179_all.deb ...
Unpacking postgresql-client-common (179) ...
Selecting previously unselected package postgresql-client-9.6.
Preparing to unpack .../2-postgresql-client-9.6_9.6.3-0ubuntu0.17.04_amd64.deb ...
Unpacking postgresql-client-9.6 (9.6.3-0ubuntu0.17.04) ...
Selecting previously unselected package postgresql-common.
Preparing to unpack .../3-postgresql-common_179_all.deb ...
Adding 'diversion of /usr/bin/pg_config to /usr/bin/pg_config.libpq-dev by postgresql-common'
Unpacking postgresql-common (179) ...
Selecting previously unselected package postgresql-9.6.
Preparing to unpack .../4-postgresql-9.6_9.6.3-0ubuntu0.17.04_amd64.deb ...
Unpacking postgresql-9.6 (9.6.3-0ubuntu0.17.04) ...
Selecting previously unselected package postgresql.
Preparing to unpack .../5-postgresql_9.6+179_all.deb ...
Unpacking postgresql (9.6+179) ...
Selecting previously unselected package postgresql-contrib-9.6.
Preparing to unpack .../6-postgresql-contrib-9.6_9.6.3-0ubuntu0.17.04_amd64.deb ...
Unpacking postgresql-contrib-9.6 (9.6.3-0ubuntu0.17.04) ...
Selecting previously unselected package sysstat.
Preparing to unpack .../7-sysstat_11.4.3-1_amd64.deb ...
Unpacking sysstat (11.4.3-1) ...
Setting up sysstat (11.4.3-1) ...

Creating config file /etc/default/sysstat with new version
update-alternatives: using /usr/bin/sar.sysstat to provide /usr/bin/sar (sar) in auto mode
Processing triggers for ureadahead (0.100.0-19) ...
Setting up libpq5:amd64 (9.6.3-0ubuntu0.17.04) ...
Processing triggers for libc-bin (2.24-9ubuntu2.2) ...
Setting up postgresql-client-common (179) ...
Processing triggers for systemd (232-21ubuntu5) ...
Setting up postgresql-common (179) ...
/usr/share/postgresql-common/supported-versions: 81: /etc/os-release: 1: parameter not set
dpkg: error processing package postgresql-common (--configure):
 subprocess installed post-installation script returned error exit status 2
Processing triggers for man-db (2.7.6.1-2) ...
Setting up postgresql-client-9.6 (9.6.3-0ubuntu0.17.04) ...
update-alternatives: using /usr/share/postgresql/9.6/man/man1/psql.1.gz to provide /usr/share/man/man1/psql.1.gz (psql.1.gz) in auto mode
dpkg: dependency problems prevent configuration of postgresql-9.6:
 postgresql-9.6 depends on postgresql-common (>= 171~); however:
  Package postgresql-common is not configured yet.

dpkg: error processing package postgresql-9.6 (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of postgresql:
 postgresql depends on postgresql-9.6; however:
  Package postgresql-9.6 is not configured yet.

dpkg: error processing package postgresql (--configure):
 dependency problems - leaving unconfigured
dpkg: dependency problems prevent configuration of postgresql-contrib-9.6:
 postgresql-contrib-9.6 depends on postgresql-9.6 (= 9.6.3-0ubuntu0.17.04); however:
  Package postgresql-9.6 is not configured yet.

dpkg: error processing package postgresql-contrib-9.6 (--configure):
 dependency problems - leaving unconfigured
No apport report written because the error message indicates its a followup error from a previous failure.
                          No apport report written because the error message indicates its a followup error from a previous failure.
                                                    No apport report written because MaxReports is reached already
                                  Processing triggers for systemd (232-21ubuntu5) ...
Processing triggers for ureadahead (0.100.0-19) ...
Errors were encountered while processing:
 postgresql-common
 postgresql-9.6
 postgresql
 postgresql-contrib-9.6
E: Sub-process /usr/bin/dpkg returned an error code (1)
```

**Steps to reproduce (if you knew)**
sudo apt install postgresql

**Expected behaviour**
PostgreSQL should be installed

**Other Note (feature-request, question, etc...)**

**Comment by jackpot51** [comment]
If this is related to the /etc/os-release modification, can you try again with the os-release file reverted to this?
```
NAME="Ubuntu"
VERSION="17.04 (Zesty Zapus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 17.04"
VERSION_ID="17.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=zesty
UBUNTU_CODENAME=zesty
```

**Comment by jliles** [comment]
Confirmed, reverting the file resolves the issue.  

It will also install after removing or moving the /etc/os-release file, however, that prevents software-properties-gtk from running.

**Comment by jackpot51** [comment]
It probably falls back to lsb release, which was not modified

**Comment by ryanleesipes** [comment]
Installed properly on current Pop!_OS. @jackpot51 can this be safely closed?

**Comment by WatchMkr** [comment]
Not installing properly with today's (7/14/17) iso.

**Comment by ryanleesipes** [comment]
@WatchMkr what do you see? Did it pop out an error?

**Comment by kylerschin** [comment]
@ryanleesipes @WatchMkr It works. Try a reinstall @jliles .

**Comment by jliles** [comment]
This does appear to be working now, after replacing the original Pop os-release file and with the current updates to Pop!_OS

**Comment by ryanleesipes** [comment]
Needs testing on a fresh install.

**Comment by kylerchin** [comment]
@jliles @ryanleesipes Pop!_OS latest version 17.04 Works!
Try updating your system at ```apt upgrade && apt dist-upgrade``` and rebooting...
```root at pop-os in /home/user/solariot/app on master*
# sudo apt install postgresql
Reading package lists... Done
Building dependency tree       
Reading state information... Done
The following packages were automatically installed and are no longer required:
  linux-headers-4.10.0-32 linux-headers-4.10.0-32-generic
  linux-image-4.10.0-32-generic linux-image-extra-4.10.0-32-generic
  linux-signed-image-4.10.0-32-generic
Use 'sudo apt autoremove' to remove them.
The following additional packages will be installed:
  libpq5 postgresql-9.6 postgresql-client-9.6 postgresql-client-common
  postgresql-common postgresql-contrib-9.6 sysstat
Suggested packages:
  postgresql-doc locales-all postgresql-doc-9.6 libdbd-pg-perl isag
The following NEW packages will be installed:
  libpq5 postgresql postgresql-9.6 postgresql-client-9.6
  postgresql-client-common postgresql-common postgresql-contrib-9.6 sysstat
0 upgraded, 8 newly installed, 0 to remove and 15 not upgraded.
Need to get 5,044 kB of archives.
After this operation, 20.3 MB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 libpq5 amd64 9.6.5-0ubuntu0.17.04 [79.7 kB]
Get:2 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 postgresql-client-common all 179 [29.0 kB]
Get:3 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 postgresql-client-9.6 amd64 9.6.5-0ubuntu0.17.04 [889 kB]
Get:4 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 postgresql-common all 179 [157 kB]
Get:5 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 postgresql-9.6 amd64 9.6.5-0ubuntu0.17.04 [3,120 kB]
Get:6 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 postgresql all 9.6+179 [5,684 B]
Get:7 http://us.archive.ubuntu.com/ubuntu zesty-updates/main amd64 postgresql-contrib-9.6 amd64 9.6.5-0ubuntu0.17.04 [491 kB]
Get:8 http://us.archive.ubuntu.com/ubuntu zesty/main amd64 sysstat amd64 11.4.3-1 [272 kB]
Fetched 5,044 kB in 3s (1,383 kB/s)
Preconfiguring packages ...
Selecting previously unselected package libpq5:amd64.
(Reading database ... 319268 files and directories currently installed.)
Preparing to unpack .../0-libpq5_9.6.5-0ubuntu0.17.04_amd64.deb ...
Unpacking libpq5:amd64 (9.6.5-0ubuntu0.17.04) ...
Selecting previously unselected package postgresql-client-common.
Preparing to unpack .../1-postgresql-client-common_179_all.deb ...
Unpacking postgresql-client-common (179) ...
Selecting previously unselected package postgresql-client-9.6.
Preparing to unpack .../2-postgresql-client-9.6_9.6.5-0ubuntu0.17.04_amd64.deb ...
Unpacking postgresql-client-9.6 (9.6.5-0ubuntu0.17.04) ...
Selecting previously unselected package postgresql-common.
Preparing to unpack .../3-postgresql-common_179_all.deb ...
Adding 'diversion of /usr/bin/pg_config to /usr/bin/pg_config.libpq-dev by postgresql-common'
Unpacking postgresql-common (179) ...
Selecting previously unselected package postgresql-9.6.
Preparing to unpack .../4-postgresql-9.6_9.6.5-0ubuntu0.17.04_amd64.deb ...
Unpacking postgresql-9.6 (9.6.5-0ubuntu0.17.04) ...
Selecting previously unselected package postgresql.
Preparing to unpack .../5-postgresql_9.6+179_all.deb ...
Unpacking postgresql (9.6+179) ...
Selecting previously unselected package postgresql-contrib-9.6.
Preparing to unpack .../6-postgresql-contrib-9.6_9.6.5-0ubuntu0.17.04_amd64.deb ...
Unpacking postgresql-contrib-9.6 (9.6.5-0ubuntu0.17.04) ...
Selecting previously unselected package sysstat.
Preparing to unpack .../7-sysstat_11.4.3-1_amd64.deb ...
Unpacking sysstat (11.4.3-1) ...
Setting up sysstat (11.4.3-1) ...

Creating config file /etc/default/sysstat with new version
update-alternatives: using /usr/bin/sar.sysstat to provide /usr/bin/sar (sar) in auto mode
Processing triggers for ureadahead (0.100.0-19) ...
ureadahead will be reprofiled on next reboot
Setting up libpq5:amd64 (9.6.5-0ubuntu0.17.04) ...
Processing triggers for libc-bin (2.24-9ubuntu2.2) ...
Setting up postgresql-client-common (179) ...
Processing triggers for systemd (232-21ubuntu5) ...
Setting up postgresql-common (179) ...
Adding user postgres to group ssl-cert

Creating config file /etc/postgresql-common/createcluster.conf with new version

Creating config file /etc/logrotate.d/postgresql-common with new version
Building PostgreSQL dictionaries from installed myspell/hunspell packages...
  en_us
Removing obsolete dictionary files:
Created symlink /etc/systemd/system/multi-user.target.wants/postgresql.service → /lib/systemd/system/postgresql.service.
Processing triggers for man-db (2.7.6.1-2) ...
Setting up postgresql-client-9.6 (9.6.5-0ubuntu0.17.04) ...
update-alternatives: using /usr/share/postgresql/9.6/man/man1/psql.1.gz to provide /usr/share/man/man1/psql.1.gz (psql.1.gz) in auto mode
Setting up postgresql-9.6 (9.6.5-0ubuntu0.17.04) ...
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Unescaped left brace in regex is deprecated, passed through in regex; marked by <-- HERE in m/(?<!\\)\${ <-- HERE ([^}]+)}/ at /usr/sbin/pam_getenv line 78.
Creating new cluster 9.6/main ...
  config /etc/postgresql/9.6/main
  data   /var/lib/postgresql/9.6/main
  locale en_US.UTF-8
  socket /var/run/postgresql
  port   5432
update-alternatives: using /usr/share/postgresql/9.6/man/man1/postmaster.1.gz to provide /usr/share/man/man1/postmaster.1.gz (postmaster.1.gz) in auto mode
Setting up postgresql (9.6+179) ...
Setting up postgresql-contrib-9.6 (9.6.5-0ubuntu0.17.04) ...
Processing triggers for systemd (232-21ubuntu5) ...
Processing triggers for ureadahead (0.100.0-19) ...
```

**Comment by kylerchin** [comment]
@jliles @ryanleesipes I would also suggest using ```apt -f install```

**Comment by jackpot51** [comment]
I believe this is fixed on 17.10 as well, since we do not change the ID in the os-release file
